Fabric.Identity.API/scripts/Install-Identity-Utilities.psm1

```diff
@@ -1625,6 +1625,305 @@ function Test-DiscoveryService{
+        [string] $azureConfigPath,
+        [Parameter(Mandatory=$true)]
+        [string] $configSection,
+        [Parameter(Mandatory=$true)]
+        [string] $childNodeGetAttribute,
+        [Parameter(Mandatory=$true)]
+        [string] $childNodeSetAttribute,
+        [Parameter(Mandatory=$true)]
+        [string] $childNodeAttributeSetting,
+        [Parameter(Mandatory=$true)]
+        [string] $childNodeAttributeValue
+    )
+
+    $azureInstallationConfig = [xml](Get-Content $azureConfigPath)
+    $azureIdentityScope = $azureInstallationConfig.installation.settings.scope | Where-Object {$_.name -eq $configSection}
+    $existingChildNode = @{}
+
+    $existingChildNode = $azureIdentityScope.ChildNodes | Where-Object {$_.$childNodeGetAttribute -eq $childNodeAttributeSetting}
+    if ($null -ne $existingChildNode)
+    {
+      $existingChildNode.$childNodeSetAttribute = $childNodeAttributeValue
+      $azureInstallationConfig.Save("$azureConfigPath") | Out-Null
+      Write-DosMessage -Level "Information" -Message "$childNodeAttributeValue added to $childNodeAttributeSetting in azureSettings.config"
+    }
+    elseif ($null -eq $existingChildNode)
+    {
+      Write-DosMessage -Level "Information" -Message "$childNodeAttributeSetting not found in azureSettings.config"
+    }
+}
+
+function Get-XMLChildNodes {
+    param (
+        [Parameter(Mandatory=$true)]
+        [string] $installConfigPath,
+        [Parameter(Mandatory=$true)]
+        [string] $configSection,
+        [Parameter(Mandatory=$true)]
+        [string[]] $nodesToSearch,
+        [Parameter(Mandatory=$true)]
+        [string] $childNodeGetAttribute
+    )
+
+    $installationConfig = [xml](Get-Content $installConfigPath)
+    $identityScope = $installationConfig.installation.settings.scope | Where-Object {$_.name -eq $configSection}
+    $allExistingChildNodes = @()
+    $count = 1
+
+    foreach($nodeToSearch in $nodesToSearch)
+    {
+      $childNodeHeader = @{}
+      $existingChildNodes = @{}
+      $setAzureSettings = $identityScope.SelectSingleNode($nodeToSearch)
+      $childNodeHeader.Add($nodeToSearch, $count)
+      $existingChildNodes = $setAzureSettings.ChildNodes | Where-Object {![string]::IsNullOrEmpty($_.$childNodeGetAttribute)}
+      $allExistingChildNodes += $childNodeHeader
+      $allExistingChildNodes += $existingChildNodes
+      if($null -eq $existingChildNodes)
+      {
+        Write-DosMessage -Level "Information" -Message "$($nodeToSearch) node or child attributes check complete"
+      }
+    }
+    return $allExistingChildNodes
+}
+
+function Check-XMLChildNode {
+    param (
+        [Parameter(Mandatory=$true)]
+        [string] $installConfigPath,
+        [Parameter(Mandatory=$true)]
+        [string] $configSection,
+        [Parameter(Mandatory=$true)]
+        [string] $nodeToSearch,
+        [Parameter(Mandatory=$true)]
+        [string] $childNodeGetAttribute,
+        [string] $childNodeGetAttribute2
```

Comment:
Comment why we need attribute and attribute2.  Use case isn't super clear for me.

Reply:
Since I wanted to make the function more generic, when checking for existing child nodes. A check on the name attribute works for the tenants, but not the registered applications, since appName could be the same but tenantId and appName is unique.